### PR TITLE
Toggle assert msg box

### PR DESCRIPTION
**Pull request by VictorStepanov**
I want to draw your attention to the new folder I file I created 
Source/Tests/MainTests.cpp

This is the new custom entry point for all of the GTests.

### CODE REVIEW

File: Source/Platform/FileWin.cpp [hunk 1]
```diff
@@ -340,10 +340,7 @@ bool Helium::IsAbsolute( const char* path )
 
 bool Helium::MakePath( const char* path )
 {
-	if (!strlen(path))
-	{
-		return false;
-	}
+	HELIUM_ASSERT(strlen(path));
```
Comment (gorlak) [inline Source/Platform/FileWin.cpp]: check [0] == '\0'

File: Source/Platform/Assert.cpp [hunk 2]
```diff
@@ -13,6 +13,7 @@
 using namespace Helium;
 
 volatile int32_t Assert::sm_active = 0;
```
Comment (gorlak) [inline Source/Platform/Assert.cpp]: I don't think either of these should be volatile.

File: Source/Platform/Assert.h [hunk 3]
```diff
@@ -77,12 +77,16 @@ namespace Helium
 		/// @name Static Utility Functions
 		//@{
 		static bool Trigger( const char* pExpression, const char* pFunction, const char* pFile, int line, const char* pMessage, ... );
+		static void SetAssertIsFatal(bool isFatal) { sm_assertIsFatal = isFatal; }
```
Comment (gorlak) [inline Source/Platform/Assert.h]: put the implementation of this in a cpp (typically i would say inl file, but this isn't high frequency enough to bother)

File: Source/Platform/Assert.h [hunk 4]
```diff
@@ -77,12 +77,16 @@ namespace Helium
...
 		//@}
 
 	private:
 		/// Non-zero if the assert handler is currently active, zero if not.
 		static volatile int32_t sm_active;
 
+		/// call abort when this is true and an assertion fails 
+		static volatile bool sm_assertIsFatal;
```
Comment (gorlak) [inline Source/Platform/Assert.h]: I'd just make this sm_fatal to keep with "when in rome"

File: Source/Platform/FilePosix.cpp [hunk 5]
```diff
@@ -340,6 +340,8 @@ bool Helium::IsAbsolute( const char* path )
 
 bool Helium::MakePath( const char* path )
 {
+	HELIUM_ASSERT(strlen(path));
```
Comment (gorlak) [inline Source/Platform/FilePosix.cpp]: check [0] == '\0'

File: Source/Platform/FileTests.cpp [hunk 6]
```diff
@@ -166,10 +166,12 @@ TEST( PlatformFileTest, PathIsNotAbsolute )
 	ASSERT_TRUE( IsAbsolute( fullPath.c_str() ) );
 }
 
+#if HELIUM_ASSERT_ENABLED 
 TEST( PlatformFileTest, FailToMakePath )
 {
-	ASSERT_FALSE(MakePath(""));
+	ASSERT_DEATH(MakePath(""), "");
```
Comment (gorlak) [inline Source/Platform/FileTests.cpp]: where is this macro coming from?  what does it do?  is it a google test macro?

File: Source/Tests/MainTests.cpp [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#include "Platform/Assert.h"
+
+#include "gtest/gtest.h"
+
+int main(int argc, char** argv) {
+	printf("Running main() from MainTests.cpp\n");
+	
+#if HELIUM_ASSERT_ENABLED 
+	Helium::Assert::SetAssertIsFatal(true);
```
Comment (gorlak) [inline Source/Tests/MainTests.cpp]: ::SetFatal, following from above

File: Source/Platform/FileWin.cpp (hunk 1 above)
Comment (VictorStepanov) [reply Source/Platform/FileWin.cpp]: done

File: Source/Platform/Assert.cpp (hunk 2 above)
Comment (VictorStepanov) [reply Source/Platform/Assert.cpp]: Yes, you are right.

File: Source/Platform/FileTests.cpp (hunk 6 above)
Comment (VictorStepanov) [reply Source/Platform/FileTests.cpp]: Yes, it is part of the GTest.

https://github.com/abseil/googletest/blob/master/googletest/docs/advanced.md#how-to-write-a-death-test

**Comment by gorlak** [comment]
👍

**Pull request merged**